### PR TITLE
feat(nepserver): add dnsserver to nepserver

### LINKED ISSUE #7: Implement a DNS-Server
https://github.com/Sn0w3y/NEP-Inverter-Reversing/tree/feature/dnsserver

@wrenix could you please have a look if it works ?

### PR DESCRIPTION
**Pull request by wrenix**
Fix #7

### CODE REVIEW

File: nepserver.py [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
+    def handle(self):
+        data, socket = self.request
+        request = DNSRecord.parse(data)
+
+        reply = DNSRecord(DNSHeader(id=request.header.id, qr=1, aa=1, ra=1), q=request.q)
+
+        if str(request.q.qname) == INTERCEPT_DOMAIN:
+            # Intercept specific domain and respond with specified IP
+            reply.add_answer(RR(request.q.qname, QTYPE.A, rdata=A(RESPONSE_IP), ttl=60))
+        else:
```
Comment (wrenix) [inline nepserver.py:30]: make it possible to disable Forward
```suggestion
        else if FORWARD_DNS not in ["", "false", "False"]:
```

File: nepserver.py [hunk 2]
```diff
@@ -141,5 +171,15 @@ def handler(*args):
             mqttc.loop_stop()
         print("Server stopped.")
 
+def run_dns_server():
+    with socketserver.ThreadingUDPServer(('0.0.0.0', 53), DNSRequestHandler) as server:
```
Comment (wrenix) [inline nepserver.py:175]: Please make the listen ip configurable (and maybe an option to disable DNS).

**Review by wrenix — COMMENTED** [review]
Looks good for the first version, i like to have 2-3 more options

**Comment by Sn0w3y** [comment]
@wrenix should be done now ?

Can you please test the functionability aswell?

Also yould you please improve the Server How to ?

File: nepserver.py [hunk 3]
```diff
@@ -15,10 +15,14 @@
...
 INTERCEPT_DOMAIN = 'www.nepviewer.net.'
 RESPONSE_IP = environ.get('RESPONSE_IP', '0.0.0.0')  # IP to respond with for intercepted domain
-FORWARD_DNS = environ.get('FORWARD_DNS', '8.8.8.8')    # DNS server to forward non-intercepted queries
+FORWARD_DNS = environ.get('FORWARD_DNS', '8.8.8.8')  # DNS server to forward non-intercepted queries
+DNS_LISTEN_ADDR = environ.get("NEP_LISTEN_ADDR", "0.0.0.0")  # IP for DNS server to listen on
+DISABLE_DNS = environ.get('DISABLE_DNS', 'False').lower() in ['true', '1', 't']
 
 class DNSRequestHandler(socketserver.BaseRequestHandler):
     def handle(self):
+        if DISABLE_DNS:
```
Comment (wrenix) [inline nepserver.py:24]: not needed it is already disabled in:
`if __name__ == "__main__":`

**Comment by wrenix** [comment]
i will test it in the evening

**Comment by Sn0w3y** [comment]
> i will test it in the evening, if it works for you - merge it and i write an bugfix (if i found a problem).

fixed the "double check" please test it as i did not test it yet :D

**Comment by wrenix** [comment]
please also/share the same Datapoint class on the `nepserver.py` then `main.py` (fix in nepserver and remove the old models.py)

**Comment by wrenix** [comment]
locks like the dns part works like expacted, i will validate with the next startup of the next inverter

**Comment by Sn0w3y** [comment]
If validated we can Merge it :) also i guess we still need to check because sometimes we get errors if defining the same IP Addresses in the variables (which makes sense)

**Comment by wrenix** [comment]
this code works for me, i would like to merge that.

What do you mean with same IP-Address?
(Every component listen on other Ports, so it should work)
I rename some environments variables and DNS- and HTTP-Server could run on different ip-address now.

**Pull request merged**